### PR TITLE
Assortment Pull Request #1, featuring TSP & P2EV Props & Keyvalues

### PR DESCRIPTION
**Pull request by Adrthegamedev**
This implements #171, with a tiny, very unfinished bit of The Stanley Parable entities. Do note that the descriptions are a bit lackluster, and i'm not all that sure about the studio model system, so look over the contraption cube props too.

**Major things:**
Started working on The Stanley Parable support, which includes:
> - `prop_tsp_button`, the only new entity that was new in the base TSP fgd itself
> - Achievements.

Added Support for P2EDU, which includes:
> - `prop_contraption_cube`
> - `prop_contraption_cube_button`
> - `point_caption_message`
> - A few new keyvalues for `trigger_catapult`
> - An assortment of new keyvalues for `func_movelinear`, `Door`, `math_remap`, and `prop_dynamic_base`

**Minor things:**
- Sets fish to baseanim, since the SDK 2013 says it is.
- Added the Talent Show achievement to Portal 2
- Mixed in a bit of Allison's P1 FGD edits; In this case, the Portal 1 soundscapes. Appropriate credit has been given in the README.MD file and as an fgd comment.

### CODE REVIEW

**Comment by TeamSpen210** [comment]
The difference between `studio()` and `studioprop()` (if that's what you mean) is that `studio()` is used for editor widgets, while `studioprop()` is used for ents that actually is that prop.

File: fgd/brush/trigger/trigger_catapult.fgd [hunk 1]
```diff
@@ -27,6 +27,10 @@
 	lowerthreshold(float) : "Lower Threshold" : 0.15 : "Flung object must be within this percentage value in order to activate fling. Specify a value between [0...1] (default is .15). This is only used if Use Threshold Check is set to yes."
 	upperthreshold(float) : "Upper Threshold" : 0.30 : "Flung object must be within this percentage value in order to activate fling. Specify a value between [0...1] (default is .30). This is only used if Use Threshold Check is set to yes."
 	entryangletolerance(float) : "Entry Angle Tolerance" : 0.0 : "Flung object's velocity must be pointing this much at the target. Specify a value between [-1...1] 1 means exactly, 0 means within 180 degrees -1 means any angle is accepted. This is only used if Use Threshold Check is set to yes."
+	
+	UseImpulses[P2EDU](float) : "Use Impulses" : "0" : "Use impulses to determine launched object velocity."
```
Comment (TeamSpen210) [inline fgd/brush/trigger/trigger_catapult.fgd:31]: I think this one should be a `bool`?

Comment (Adrthegamedev) [reply fgd/brush/trigger/trigger_catapult.fgd:31]: You could be right. I only used "float" because the edu FGD itself uses float for it.

**Comment by Adrthegamedev** [comment]
> The difference between `studio()` and `studioprop()` (if that's what you mean) is that `studio()` is used for editor widgets, while `studioprop()` is used for ents that actually is that prop.

Then, some changes might have to be made. I used the base "studio" attributes for the studioprops of prop_contraption_cube (the reflection cube model) and prop_contraption_cube_button (the default box button model), not knowing that that's meant to actually be the item.

**Comment by Luke18033** [comment]
Specifically the difference between `studio` and `studioprop` is that the latter will resize the bounding box to match the entity's rotation, while the former won't. Generally I'd suggest `studioprop` for anything where the bounding box would be unevenly sized by default.

**Comment by Luke18033** [comment]
Also, I'd suggest cleaning up the FGDs a bit, currently they seem to just be direct copy-pastes of Valve's entries. For example the contraption cube friction and elasticity should probably be `choices` type, with the dropdown including the 5 options for each.

As for TSP, I know there are a few other entities that it doesn't have like `weapon_portalgun`, and it *does* have `weapon_paintgun` weirdly enough. You probably want to check `dumpentityfactories` to see exactly what's different.

**Comment by wertercatt** [comment]
> You probably want to check `dumpentityfactories` to see exactly what's different.

https://gist.github.com/wertercatt/3d4a2304e3cf20f400643ffbbfce198e#file-portal2-vs-thestanleyparable-diff-L246

Removed:
```
challenge_mode_end_node
env_fizzler_effects
env_portal_laser
fizzler_multiorigin_sound_player
func_placement_clip
info_paint_sprayer
item_nugget
npc_portal_turret_floor
paint_sphere
paint_stream
portal_gamerules
portalmp_gamerules
prop_paint_bomb
prop_physics_paintable
trigger_paint_cleanser
weapon_portalgun
weapon_promo_antenna_ball
weapon_promo_antenna_egg
weapon_promo_helmet_ball
weapon_promo_helmet_egg
```
added:
```
prop_tsp_button
trigger_still
weapon_paintgun
weapon_smg1
```

**Comment by Adrthegamedev** [comment]
I (unfortunately) only have the demo version of TSP (meaning i can't peek into the maps to see how some of the ents are used), but I did test a few things out there, using Hammer. `trigger_still` seems to act like a regular trigger ent, expectedly.

**Comment by TeamSpen210** [comment]
Can you turn on "allow edits from maintainers" on the right, so I can push to your branch?

**Comment by Adrthegamedev** [comment]
> Can you turn on "allow edits from maintainers" on the right, so I can push to your branch?

It's already on for me.

**Comment by Adrthegamedev** [comment]
@TeamSpen210 I've now invited ye to the repo, hopefully that helps

**Comment by TeamSpen210** [comment]
Ah figured out the problem, it's because you're committing on master not a separate branch.

**Pull request merged**